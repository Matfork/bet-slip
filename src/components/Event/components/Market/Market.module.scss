@import "src/assets/scss/media";
@import "src/assets/scss/spacing";
@import "src/assets/scss/colors";

.Market-component {
  .Market-component__container {
    margin: $spacing-2 0;

    .MuiDivider-root {
      margin-top: $spacing-2;
    }

    &:first-child {
      margin-top: 0;
    }

    &:last-child {
      margin-bottom: 0;

      .MuiDivider-root {
        display: none;

        &:hover.Mui-disabled {
          background-color: $green !important;
          color: $white !important;
        }
      }
    }

    .Market-component__label {
      h6 {
        margin-bottom: $spacing-2;
      }
    }
  }
}
