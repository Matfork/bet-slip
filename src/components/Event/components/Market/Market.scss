@import "src/assets/scss/media";
@import "src/assets/scss/spacing";

.Market-component {
  .Market-component__container {
    margin: $spacing-2 0;

    .MuiDivider-root {
      margin-top: $spacing-2;
    }

    &:first-child {
      margin-top: 0;
    }

    &:last-child {
      margin-bottom: 0;

      .MuiDivider-root {
        display: none;
      }
    }

    .Market-component__label {
      h6 {
        margin-bottom: $spacing-2;
      }
    }
  }
}
