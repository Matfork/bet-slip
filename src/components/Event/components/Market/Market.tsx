import { Divider, Typography } from '@material-ui/core';
import React from 'react';
import { MarketType } from 'domains/market';
import Selection from '../Selection/Selection';
import styles from './Market.module.scss';

interface MarketProps {
  markets: MarketType[];
}

const className = 'Market-component';

const MarketComponent: React.FC<MarketProps> = ({ markets }: MarketProps) => {
  return (
    <div className={styles[className]}>
      {markets.map((market) => (
        <div key={market.id} className={styles[`${className}__container`]}>
          <div className={styles[`${className}__label`]}>
            <Typography variant="h6">{market.name}</Typography>
          </div>
          <div className={styles[`${className}__options`]}>
            <Selection selections={market.selections} />
          </div>
          <Divider className={styles[`MuiDivider-root`]} />
        </div>
      ))}
    </div>
  );
};

export default MarketComponent;
