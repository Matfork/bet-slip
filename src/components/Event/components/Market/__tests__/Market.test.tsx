import { screen } from '@testing-library/react';
import { customRender } from 'test.utilities';
import { marketsStub } from 'test.stub';
import Market from '../Market';

describe('Market Component', () => {
  it('should test snapshot', () => {
    const component = customRender(<Market markets={marketsStub} />);
    expect(component.container).toMatchSnapshot();
  });

  it('should display all markets avaiable', () => {
    customRender(<Market markets={marketsStub} />);
    const titleEl = screen.queryByText(/Team to Win/);
    expect(titleEl).toBeInTheDocument();
  });

  it('should not display events if there is no data', () => {
    customRender(<Market markets={[]} />);
    const titleEl = screen.queryByText(/Team to Win/);
    expect(titleEl).toBeNull();
  });
});
