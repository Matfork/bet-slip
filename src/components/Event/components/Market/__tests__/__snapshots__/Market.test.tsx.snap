// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Market Component should test snapshot 1`] = `
<div>
  <div
    class="Market-component"
  >
    <div
      class="Market-component__container"
    >
      <div
        class="Market-component__label"
      >
        <h6
          class="MuiTypography-root MuiTypography-h6"
        >
          Team to Win
        </h6>
      </div>
      <div
        class="Market-component__options"
      >
        <div
          class="Selection-component"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-outlined Selection-component__button  MuiButton-outlinedSizeSmall MuiButton-sizeSmall MuiButton-disableElevation"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              <p
                class="MuiTypography-root MuiTypography-body2"
              >
                Real Madrid
              </p>
              <p
                class="MuiTypography-root MuiTypography-body2"
              >
                1.23
              </p>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-outlined Selection-component__button  MuiButton-outlinedSizeSmall MuiButton-sizeSmall MuiButton-disableElevation"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              <p
                class="MuiTypography-root MuiTypography-body2"
              >
                Barcelona
              </p>
              <p
                class="MuiTypography-root MuiTypography-body2"
              >
                2.05
              </p>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
      <hr
        class="MuiDivider-root MuiDivider-root"
      />
    </div>
    <div
      class="Market-component__container"
    >
      <div
        class="Market-component__label"
      >
        <h6
          class="MuiTypography-root MuiTypography-h6"
        >
          Player to Score First
        </h6>
      </div>
      <div
        class="Market-component__options"
      >
        <div
          class="Selection-component"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-outlined Selection-component__button  MuiButton-outlinedSizeSmall MuiButton-sizeSmall MuiButton-disableElevation"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              <p
                class="MuiTypography-root MuiTypography-body2"
              >
                Ronaldo
              </p>
              <p
                class="MuiTypography-root MuiTypography-body2"
              >
                1.15
              </p>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-outlined Selection-component__button  MuiButton-outlinedSizeSmall MuiButton-sizeSmall MuiButton-disableElevation"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              <p
                class="MuiTypography-root MuiTypography-body2"
              >
                Messi
              </p>
              <p
                class="MuiTypography-root MuiTypography-body2"
              >
                1.3
              </p>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-outlined Selection-component__button  MuiButton-outlinedSizeSmall MuiButton-sizeSmall MuiButton-disableElevation"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              <p
                class="MuiTypography-root MuiTypography-body2"
              >
                Bale
              </p>
              <p
                class="MuiTypography-root MuiTypography-body2"
              >
                1.35
              </p>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
      <hr
        class="MuiDivider-root MuiDivider-root"
      />
    </div>
  </div>
</div>
`;
